### PR TITLE
feat(simulator): export simulator UDIDs, wait-until-simulator-booted command

### LINKED ISSUE #17: Make Simulator UUIDs visible to subsequent jobs
## Describe Request:

When running certain commands against the simulator(s) it's sometimes necessary to ensure they have fully started.
The current semi-documented method is described [on stackoverflow][1]

For this to work, the UUIDs of the simulator(s) need to be visible to subsequent tasks.
According to the [docs][2] this is best achieved by exporting the variable to `BASH_ENV`

The variable name could be parameterised, with a sane default.

The suggestion is to add the UUID of the simulator device(s) to `BASH_ENV` for subsequent ease of use.

## Examples:

```yaml
---
version: 2.1

orbs:
  macos: circleci/macos@2.1.0

jobs:
  ios_tests:
    macos:
      xcode: '13.2.1'
    resource_class: 'medium'
    steps:
      - checkout
      - macos/preboot-simulator:
          device: 'iPhone 11'
          platform: 'iOS'
          version: '15.2'
          device_id_var: 'IPHONE_UUID'    # new parameter
     - run: xcrun simctl bootstatus ${IPHONE_UUID}  # ensures iPhone is fully booted
     - run: some command that requires the simulator fully booted
```

## Supporting Documentation Links:
[1]: https://stackoverflow.com/questions/52438954/mac-simulator-xcode-wait-for-startup-command
[2]: https://circleci.com/docs/2.0/env-vars/#using-parameters-and-bash-environment

### PR DESCRIPTION
**Pull request by bdellegrazie**
* Adds the capability to export UDIDs of simulators for subsequent commands to use
* Adds a wait-until-simulator-booted command

Fixes #17

### CODE REVIEW

**Comment by felicianotech** [comment]
Looks good to me, @BytesGuy ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
  - Current development changes [ to be moved to release ]
+ - Export Device UDIDs when pre-booting simulators
+ - Added wait-until-simulator-booted command
```
Comment (BytesGuy) [inline CHANGELOG.md:10]: This can be removed as I will bump up the orb version when this PR is merged

File: src/commands/preboot-simulator.yml [hunk 2]
```diff
@@ -4,6 +4,10 @@ parameters:
     description: The device type. E.g., "iPhone 12"
     type: string
     default: "iPhone 12"
+  device-udid-var:
+    description: The name of the variable to contain the device's UDID e.g. IPHONE_UDID
+    type: string
+    default: DEVICE_UDID
```
Comment (BytesGuy) [inline src/commands/preboot-simulator.yml:10]: Let's make this something like `MACOS_ORB_DEVICE_UDID` by default. This just makes it explicit as to what the variable is and will prevent any possible collisions with users existing commands/scripts

File: src/commands/preboot-paired-simulator.yml [hunk 3]
```diff
@@ -4,18 +4,31 @@ parameters:
     description: The device type. E.g., "iPhone 13 Pro Max"
     type: string
     default: "iPhone 13 Pro Max"
+  iphone-device-udid-var:
+    description: The name of the variable to contain the iphone's UDID e.g. IPHONE_UDID
+    type: string
+    default: IPHONE_UDID
```
Comment (BytesGuy) [inline src/commands/preboot-paired-simulator.yml:10]: Same as my comment in the other file, let's make these something like `MACOS_ORB_<type>_UDID` by default

**Review by BytesGuy — CHANGES_REQUESTED** [review]
Thanks for this @bdellegrazie and sorry it took us a while to get round to the review. I've requested a few changes which will make this PR ready to merge. Thanks :)

File: CHANGELOG.md (hunk 1 above)
Comment (bdellegrazie) [reply CHANGELOG.md]: done

File: src/commands/preboot-simulator.yml (hunk 2 above)
Comment (bdellegrazie) [reply src/commands/preboot-simulator.yml]: done

File: src/commands/preboot-paired-simulator.yml (hunk 3 above)
Comment (bdellegrazie) [reply src/commands/preboot-paired-simulator.yml]: done

**Comment by bdellegrazie** [comment]
@BytesGuy I have updated as requested, please re-review at your leisure

**Comment by bdellegrazie** [comment]
@BytesGuy any chance of a re-review please?

File: src/commands/preboot-paired-simulator.yml [hunk 4]
```diff
@@ -26,8 +39,11 @@ steps:
           ESCAPED_WATCHOS_VER=$(echo << parameters.watch-version >> | tr '.' '-')
           SIMLIST=$(xcrun simctl list -j)
           IPHONE_UDID=$(echo $SIMLIST | jq -r ".devices.\"com.apple.CoreSimulator.SimRuntime.iOS-$ESCAPED_IPHONE_VER\"[] | select(.name==\"<< parameters.iphone-device >>\").udid")
+          echo "export <<parameters.iphone-udid-var>>=$IPHONE_UDID" >> $BASH_ENV
```
Comment (EricRibeiro) [inline src/commands/preboot-paired-simulator.yml:42]: The parameter `iphone-udid-var` is referenced here but it is never declared. Maybe you meant: `iphone-device-udid-var` ?

File: src/commands/preboot-paired-simulator.yml [hunk 5]
```diff
@@ -26,8 +39,11 @@ steps:
...
           WATCH_UDID=$(echo $SIMLIST | jq -r ".devices.\"com.apple.CoreSimulator.SimRuntime.watchOS-$ESCAPED_WATCHOS_VER\"[] | select(.name==\"<< parameters.watch-device >>\").udid")
+          echo "export <<parameters.watch-udid-var>>=$WATCH_UDID" >> $BASH_ENV
```
Comment (EricRibeiro) [inline src/commands/preboot-paired-simulator.yml:44]: The parameter `watch-udid-var` is referenced here but it is never declared. Maybe you meant: `watch-device-udid-var` ?

**Review by EricRibeiro — CHANGES_REQUESTED** [review]

File: src/commands/preboot-paired-simulator.yml (hunk 4 above)
Comment (bdellegrazie) [reply src/commands/preboot-paired-simulator.yml]: I did, thank you - fixed.

File: src/commands/preboot-paired-simulator.yml (hunk 5 above)
Comment (bdellegrazie) [reply src/commands/preboot-paired-simulator.yml]: I did, thank you - fixed.

**Comment by bdellegrazie** [comment]
@EricRibeiro please re-review

**Comment by bdellegrazie** [comment]
(rebased on main too)

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/macos?version=dev:710676e4b1a35c0667283684bcc867f4dbb43044

**Review by BytesGuy — APPROVED** [review]

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/macos?version=dev:cb10a6260c18c5741fb41ed26534d3190e941c74

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/macos?version=dev:cb10a6260c18c5741fb41ed26534d3190e941c74

**Comment by orb-publisher** [comment]
Your orb has been published to the CircleCI Orb Registry.
You can view your published orb on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/macos?version=2.2.0

**Pull request merged**